spec/forms/savings_amounts/savings_amounts_form_spec.rb

```diff
@@ -106,20 +106,33 @@
       end
     end
 
-    context 'check boxes are unchecked' do
-      let(:check_box_params) { check_box_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = '' } }
+    context 'some check boxes are unchecked' do
+      let(:check_box_params) do
+        boxes = check_box_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = '' }
+        boxes[:check_box_isa] = 'true'
+        boxes
+      end
 
-      shared_examples_for 'it empties amounts' do
-        it 'empties amounts' do
+      context 'amounts are valid' do
+        let(:amount_params) { attributes.each_with_object({}) { |attr, hsh| hsh[attr] = Faker::Number.decimal.to_s } }
+
+        it 'empties amounts if checkbox is unchecked' do
+          attributes_except_isa = attributes - [:isa]
           subject.save
           savings_amount.reload
-
-          attributes.each do |attr|
+          attributes_except_isa.each do |attr|
             val = savings_amount.send(attr)
```

Comment:
Not you're change, but as you're here .... `__send__` is preferred to `send`. So:
```ruby
val = savings_amount.__send__(attr)
```